backend/fms_core/services/experiment_run.py

```diff
@@ -0,0 +1,170 @@
+    blank_str_to_none,
+    float_to_decimal,
+    get_normalized_str,
+)
+
+def create_experiment_run(experiment_type, instrument, container, start_date,
+                 sample_rows_for_experiment, properties,
+                 protocols_objs_dict, property_types_objs_dict):
+    errors = {}
+    experiment_run = None
+
+    parent_process, processes_by_protocol_id = create_processes_from_protocols(protocols_objs_dict)
+
+    try:
+        experiment_run = ExperimentRun.objects.create(experiment_type=experiment_type,
+                                     instrument=instrument,
+                                     container=container,
+                                     process=parent_process,
+                                     start_date=start_date)
+
+        print('SERVICES - experiment_run: ', experiment_run)
+
+        result = create_properties(properties, property_types_objs_dict, processes_by_protocol_id)
+        if result['errors'] != {}:
+            errors['properties'] = result['errors']
+
+        print('SERVICES - properties/result: ', result)
+
+        result = handle_samples(experiment_run, sample_rows_for_experiment)
+        if result['errors'] != {}:
+            errors['samples'] = result['errors']
+
+        print('SERVICES - samples/result: ', result)
+
+    except Exception as e:
+        errors['experiment_run'] = e
+        print('SERVICES - experiment_run/exception: ', e)
+
+
+    return {'object': experiment_run, 'errors': errors}
+
+def create_processes_from_protocols(protocols_objs_dict):
+    processes_by_protocol_id = {}
+    for protocol in protocols_objs_dict.keys():
+        parent_process = Process.objects.create(protocol=protocol, comment="")
+        for subprotocol in protocols_objs_dict[protocol]:
+            sp = Process.objects.create(protocol=subprotocol,
+                                        parent_process=parent_process,
+                                        comment="Experiment")
+            processes_by_protocol_id[subprotocol.id] = sp
+
+    return (parent_process, processes_by_protocol_id)
+
+
+def create_properties(properties, property_types_objs_dict, processes_by_protocol_id):
+    errors = {}
+    # Create property values for ExperimentRun
+    for i, (property, value) in enumerate(properties.items()):
+        property_type = property_types_objs_dict[property]
+        protocol_id = property_type.object_id
+        process = processes_by_protocol_id[protocol_id]
+
+        if type(value).__name__ in ('datetime', 'time'):
+            value = value.isoformat().replace("T00:00:00", "")
+
+        try:
+            PropertyValue.objects.create(value=str(value),
+                                         property_type=property_type,
+                                         content_object=process)
+        except Exception as e:
+            errors[property] = e.error_dict['value']
+
+    return {'errors': errors}
+
+
+def handle_samples(experiment_run, samples):
```

Comment:
Handle is often a term used to described an ID that point to a an object ... Handle is also a very general term to describe pretty much anything we do on something. Is there a more precise way to describe what is happening here ?